elyra/api/elyra.yaml

```diff
@@ -490,6 +519,41 @@ components:
         overwrite:
           type: boolean
           description: Determines if the existing export should be overwritten.
+          
+    ResourceBody:
+      description: The set of resources comprising the request body.
+      required:
+        - env_vars
+        - inputs
+        - outputs
+      type: object
+      properties:
+        env_vars:
+          type: object
+          properties: 
+            ENV_VAR_NAME1:
+              type: string
+            ENV_VAR_NAME2:
+              type: string
```

Comment:
Each of the properties (`env_vars`, `inputs`, `outputs`) is a list, correct?  This should equate to type `array`.

Since `env_vars` are a list of strings, `env_vars:` should be expressed as something like:
```
env_vars:
    type: array
    items: string
```
I suspect the others (`inputs` and `outputs`) could be similarly defined as well.

Reply:
Each property is actually it's own dictionary, since we also return the value if application. Here's an example from a real request I just made:

```
{
    "env_vars": {
        "DATASET_URL": "https://dax-cdn.cdn.appdomain.cloud/dax-noaa-weather-data-jfk-airport/1.1.4/noaa-weather-data-jfk-airport.tar.gz"
    },
    "inputs": {},
    "outputs": {}
}
```

Reply:
I was thinking about `inputs` and `outputs` while I was putting this together actually. Since they won't really be key:value pairs like environment variables can be, I think we'll eventually want to change them to arrays rather than dictionaries. But for now our implementation is an empty dictionary for each.

Reply:
Ok - thanks for the information.  I think, then, we want to convey these are dictionaries of strings and it appears that is done [this way](https://swagger.io/docs/specification/data-models/dictionaries/):

```
env_vars:
    type: object
    additionalProperties:
         type: string
```

Regarding the `inputs` and `outputs` entries, if we know they'll be lists of strings, I think it might be a good idea to update the implementation to return an empty list, otherwise, we could be looking at an incompatibility - both with "auto-processors" of this document but also with our frontend devs since they may want to populate results - albeit empty - and should use the correct data type.

Reply:
Sounds good! I'm happy to update the implementation as well -- would we want that in a separate PR?

Reply:
I think updating the implementation in this PR is fine, especially since there aren't any consumers yet.

Reply:
Makes sense!